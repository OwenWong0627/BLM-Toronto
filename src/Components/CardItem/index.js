import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

/**
 * This component represents a single card item that will be rendered on the home page, in the Cards component
 * This component will accept a prop, and this component will destructure it to extract information from the prop
 * to display the information in the card item
 */
function CardItem(props) {
   return (
      <>
         <li className='cards__item'>
            <Link to={props.to} data-testid='card-item' className='cards__item__link'>
               <figure data-testid='card-item-label' className='cards__item__pic-wrap' data-category={props.label}>
                  <img
                     data-testid='card-item-img'
                     className='cards__item__img'
                     alt='Page Previews'
                     src={props.src}
                  />
               </figure>
               <div className='cards__item__info'>
                  <h2 className='cards__item__text'>{props.text}</h2>
               </div>
            </Link>
         </li>
      </>
   );
}

/**
 * propType Documentation
 */
CardItem.propTypes = {
   to: PropTypes.string.isRequired,
   label: PropTypes.string.isRequired,
   src: PropTypes.string.isRequired,
   text: PropTypes.string.isRequired,
}

export default CardItem;