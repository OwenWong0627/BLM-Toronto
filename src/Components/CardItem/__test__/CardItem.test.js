import React from 'react';
import ReactDOM from 'react-dom';
import CardItem from '..';

import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Router><CardItem src="/someImage.png" text="sample-text" label="text" to='/sample-link' /></Router>, div);
});

it('Renders a card item correctly', () => {
   const { getByTestId } = render(<Router><CardItem
      src='http://ctt.trains.com/sitefiles/images/no-preview-available.png'
      text='Click for Map'
      label='Map'
      to='/map'
   /></Router>);
   expect(getByTestId('card-item')).toHaveTextContent("Click for Map");
   expect(getByTestId('card-item')).toHaveAttribute("href", "/map");
   expect(getByTestId('card-item-label')).toHaveAttribute("data-category", "Map");
   expect(getByTestId('card-item-img')).toHaveAttribute("src", "http://ctt.trains.com/sitefiles/images/no-preview-available.png");
});

it('matches snapshot', () => {
   const tree = renderer.create(<Router><CardItem
      src='http://ctt.trains.com/sitefiles/images/no-preview-available.png'
      text='Click for Map'
      label='Map'
      to='/map'
   /></Router>).toJSON();
   expect(tree).toMatchSnapshot();
});