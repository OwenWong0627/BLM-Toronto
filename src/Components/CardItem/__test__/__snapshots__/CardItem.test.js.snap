// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  className="cards__item"
>
  <a
    className="cards__item__link"
    data-testid="card-item"
    href="/map"
    onClick={[Function]}
  >
    <figure
      className="cards__item__pic-wrap"
      data-category="Map"
      data-testid="card-item-label"
    >
      <img
        alt="Page Previews"
        className="cards__item__img"
        data-testid="card-item-img"
        src="http://ctt.trains.com/sitefiles/images/no-preview-available.png"
      />
    </figure>
    <div
      className="cards__item__info"
    >
      <h2
        className="cards__item__text"
      >
        Click for Map
      </h2>
    </div>
  </a>
</li>
`;
